.github/workflows/python_tests.yml

```diff
@@ -18,15 +18,26 @@ jobs:
       - uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-      # https://lxml.de/installation.html#requirements
-      - run: sudo apt-get install libxml2-dev libxslt-dev
-      - run: pip install --upgrade pip setuptools wheel
-      - run: pip install -r requirements_test.txt
-      - run: pip list --outdated
-      - run: make lint-diff
-      - run: make git
-      - run: make i18n
-      - run: make lint
-      - run: make test-py
-      - run: source scripts/run_doctests.sh
-      - run: mypy .
+      - uses: actions/cache@v2
+        with:
+          path: ~/openlibrary_python_venv
```

Comment:
It looks like we can avoid using a virtual env if we cache `~/.cache/pip` ( https://github.com/actions/cache/blob/main/examples.md#python---pip ). That removes a layer of abstraction, so if it works, it would be nice :)

Reply:
Yes, that is simple but that only caches the downloads, the installation step will still take place. As recommended by @anandology in [this comment](https://github.com/internetarchive/infogami/pull/161#issuecomment-814651787), we went with using the virtual environment in the CI and caching that directly. That way, we only need to update the virtual environment when any version changes happen in our dependencies file. It's a bit of a boost seeing the number of dependencies we have :)